src/ch01-00-foreword.md

```diff
@@ -0,0 +1,39 @@
+# Foreword
+
+In recent years, it has become an industry standard to deploy software using
+some form of containerization; and while this trend has helped to reduce the
+complexity of our deployments by offering process and file isolation at
+runtime, it has done virtually nothing to reduce the complexity of building
+and maintaining software.
+
+Nix tries to bridge this gap using the same isolation primitizes used by
+container technologies to ensure environmental conditions are precisely
+mirrored _at build time_.
+
+Thanks to it's powerful expression language, Nix's ability to compose packages
+is versatile enough even to provide elegant solutions for generating these same
+OCI compliant containers in a totally duplicable fashion.
```

Comment:
```suggestion
OCI compliant containers in a totally DRY fashion.
```

Reply:
not sure if the requested change actually has the intended meaning. If we build a container from a dockerfile several times, we are not really repeating ourselves(not modifying the file), but we are also not guaranteed to get the same container every run either.

Reply:
when I look back to when I started refactoring dockerfiles into nix, it was because I could derive several images with ease from a single shared config state.

That was real nice, since it made it a lot easier to keep two images that are part of a microservice architecture in sync.

I also was intrigued by the relatively lean images.

Reply:
I mean people had invented ksonnet for some of those composability reasons... https://github.com/ksonnet/ksonnet/tree/master/lib

Reply:
Yes well, I have personally never been a fan of docker, despite having to uses under certain conditions, so if you'd like to improve this section, then please do so, because I feel my lack of experience coming off :sweat_smile: